drivers/sensor/adltc2990/adltc2990.c

```diff
@@ -0,0 +1,470 @@
+	return status_reg & BIT(0);
+}
+
+static void adltc2990_get_v1_v2_val(const struct device *dev, struct sensor_value *val,
+				    uint8_t num_values, uint8_t *const offset_index)
+{
+	struct adltc2990_data *data = dev->data;
+
+	for (uint8_t index = 0; index < num_values; index++) {
+		val[index].val1 = data->pins_v1_v2_values[index] / 1000000;
+		val[index].val2 = data->pins_v1_v2_values[index] % 1000000;
+		*offset_index = index + 1;
+	}
+}
+
+static void adltc2990_get_v3_v4_val(const struct device *dev, struct sensor_value *val,
+				    uint8_t num_values, uint8_t const *const offset)
+{
+	struct adltc2990_data *data = dev->data;
+
+	uint8_t offset_index = *offset;
+
+	for (uint8_t index = 0; index < num_values; index++) {
+		val[index + offset_index].val1 = data->pins_v3_v4_values[index] / 1000000;
+		val[index + offset_index].val2 = data->pins_v3_v4_values[index] % 1000000;
+	}
+}
+
+static int adltc2990_trigger_measurement(const struct device *dev)
+{
+	const struct adltc2990_config *cfg = dev->config;
+
+	return i2c_reg_write_byte_dt(&cfg->bus, ADLTC2990_REG_TRIGGER, 0x1);
+}
+
+static int32_t adltc2990_get_property_value(const struct device *dev,
+					    enum adltc2990_monitoring_type type,
+					    enum adltc2990_monitor_pins pin)
+{
+	const struct adltc2990_config *cfg = dev->config;
+
+	uint8_t msb_value = 0, lsb_value = 0;
+	uint8_t msb_address, lsb_address;
+
+	switch (pin) {
+	case V1: {
+		msb_address = ADLTC2990_REG_V1_MSB;
+		lsb_address = ADLTC2990_REG_V1_LSB;
+		break;
+	}
+	case V2: {
+		msb_address = ADLTC2990_REG_V2_MSB;
+		lsb_address = ADLTC2990_REG_V2_LSB;
+		break;
+	}
+	case V3: {
+		msb_address = ADLTC2990_REG_V3_MSB;
+		lsb_address = ADLTC2990_REG_V3_LSB;
+		break;
+	}
+	case V4: {
+		msb_address = ADLTC2990_REG_V4_MSB;
+		lsb_address = ADLTC2990_REG_V4_LSB;
+		break;
+	}
+	case INTERNAL_TEMPERATURE: {
+		msb_address = ADLTC2990_REG_INTERNAL_TEMP_MSB;
+		lsb_address = ADLTC2990_REG_INTERNAL_TEMP_LSB;
+		break;
+	}
+	case SUPPLY_VOLTAGE: {
+		msb_address = ADLTC2990_REG_VCC_MSB;
+		lsb_address = ADLTC2990_REG_VCC_LSB;
+		break;
+	}
+	default: {
+		LOG_ERR("Trying to access illegal register");
+		return -EINVAL;
+	}
+	}
+
+	i2c_reg_read_byte_dt(&cfg->bus, msb_address, &msb_value);
+	i2c_reg_read_byte_dt(&cfg->bus, lsb_address, &lsb_value);
+	uint16_t conversion_factor;
+	uint8_t negative_bit_index;
+
+	if (type == VOLTAGE_SINGLEENDED) {
+		conversion_factor = ADLTC2990_VOLTAGE_SINGLEENDED_CONVERSION_FACTOR;
+		negative_bit_index = 14;
+	} else if (type == VOLTAGE_DIFFERENTIAL) {
+		conversion_factor = ADLTC2990_VOLTAGE_DIFFERENTIAL_CONVERSION_FACTOR;
+		negative_bit_index = 14;
+	} else if (type == TEMPERATURE) {
+		conversion_factor = ADLTC2990_TEMPERATURE_CONVERSION_FACTOR;
+		negative_bit_index = 12;
+	} else {
+		LOG_ERR("unknown type");
+		return -EINVAL;
+	}
+
+	int16_t value = (msb_value<<8)+lsb_value;
+
+	int32_t voltage_value = (value << (31-negative_bit_index))>>(31-negative_bit_index);
+
```

Comment:
In response to : https://github.com/zephyrproject-rtos/zephyr/pull/60842#discussion_r1323581251
Hi @MaureenHelm, i guess this is how you meant it, or?

Reply:
Yes